scarb/src/bin/scarb/commands/fmt.rs

```diff
@@ -1,21 +1,23 @@
 use anyhow::Result;
 
 use crate::args::FmtArgs;
+use crate::errors::ErrorWithExitCode;
 use scarb::core::Config;
 use scarb::ops;
 
 #[tracing::instrument(skip_all, level = "info")]
 pub fn run(args: FmtArgs, config: &Config) -> Result<()> {
     let ws = ops::read_workspace(config.manifest_path(), config)?;
-    match ops::format(
+    if ops::format(
         ops::FmtOptions {
             check: args.check,
             pkg_name: args.package,
             color: !args.no_color,
         },
         &ws,
-    ) {
-        Ok(true) => Ok(()),
-        _ => std::process::exit(1),
+    )? {
+        Ok(())
+    } else {
+        Err(ErrorWithExitCode::code(1).into())
```

Comment:
wdyt about having such helper?

```suggestion
        error_with_exit_code(1)
```